pkg/controller/provider/openstack/handler.go

```diff
@@ -38,6 +39,7 @@ type Handler struct {
 
 var _ provider.DNSHandler = &Handler{}
 
+// NewHandler constructs a new DNSHandler object.
 func NewHandler(config *provider.DNSHandlerConfig) (provider.DNSHandler, error) {
 	authConfig, err := readAuthConfig(config)
```

Comment:
There is no need to use the `readAuthConfig` function, since `clientconfig.AuthOptions` already fetches env parameters. I also don't see a reason why do you need to keep the `authConfig` struct. Take a look on this branch: https://github.com/kayrus/external-dns-management/tree/new-auth

Reply:
Thanks, I missed this part. Nevertheless, I don't like the current situation with the auth options. It looks even worse than in https://github.com/kubernetes/cloud-provider-openstack. I propose to create a package, which will handle all the auth options in one place.

Reply:
That's true, it looks a bit complicated now. I suggest to apply these changes to the pull request:
https://github.com/gardener/external-dns-management/commit/88eeb9687f10ae62c7044969067922938c133f9d

Reply:
Can we keep that for now as is?